jenkins/spark-premerge-build.sh

```diff
@@ -37,27 +37,41 @@ mvn_verify() {
     # file size check for pull request. The size of a committed file should be less than 1.5MiB
     pre-commit run check-added-large-files --from-ref $BASE_REF --to-ref HEAD
 
+    # Get Spark versions
+    SPARK_VERSIONS_SNAPSHOTS_STR=$(mvn -B help:evaluate -q -pl dist -Psnapshots -Dexpression=included_buildvers -DforceStdout)
+    SPARK_VERSIONS_NOSNAPSHOTS_STR=$(mvn -B help:evaluate -q -pl dist -PnoSnapshots -Dexpression=included_buildvers -DforceStdout)
+    SPARK_VERSIONS_SNAPSHOTS_STR=$(echo $SPARK_VERSIONS_SNAPSHOTS_STR)
+    SPARK_VERSIONS_NOSNAPSHOTS_STR=$(echo $SPARK_VERSIONS_NOSNAPSHOTS_STR)
+    oldIFS=$IFS
+    IFS=$', ' <<< $SPARK_VERSIONS_SNAPSHOTS_STR read -r -a SPARK_SHIM_VERSIONS
+    IFS=$', ' <<< $SPARK_VERSIONS_NOSNAPSHOTS_STR read -r -a SPARK_SHIM_VERSIONS_NOSNAPSHOTS
+    IFS=$oldIFS
+    SPARK_SHIM_VERSIONS_TEST=("311" "320" "330")
```

Comment:
it feels a bit odd to have this here when everything else comes from pom file.  If we remove 311 then we have to come modify here, maybe we should move this to pom file property

Reply:
We don't have to hardcode it if we sort based on the first two digits and output the first element of each run.
```bash
$ echo "311
312
320
321" | sort -n -k1.1,1.2 -u -s
311
320
```